addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQCloudHandler.java

```diff
@@ -0,0 +1,277 @@
+import org.syphr.sleepiq.api.UnauthorizedException;
+import org.syphr.sleepiq.api.model.Bed;
+import org.syphr.sleepiq.api.model.BedStatus;
+import org.syphr.sleepiq.api.model.FamilyStatus;
+
+import com.google.common.base.Objects;
+
+/**
+ * The {@link SleepIQCloudHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gregory Moyer - Initial contribution
+ */
+public class SleepIQCloudHandler extends ConfigStatusBridgeHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPE_UIDS = Collections.singleton(THING_TYPE_CLOUD);
+
+    private final Logger logger = LoggerFactory.getLogger(SleepIQCloudHandler.class);
+
+    private List<BedStatusListener> bedStatusListeners = new CopyOnWriteArrayList<>();
+
+    private ScheduledFuture<?> pollingJob;
+
+    private Runnable pollingRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            publishBedStatusUpdates();
+        }
+    };
+
+    private SleepIQ cloud;
+
+    public SleepIQCloudHandler(final Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+
+        try {
+            createCloudConnection();
+
+            logger.debug("Setting SleepIQ cloud online");
+            updateListenerManagement();
+            updateStatus(ThingStatus.ONLINE);
+
+        } catch (UnauthorizedException e) {
+            logger.error("SleepIQ cloud authentication failed", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid SleepIQ credentials");
+        } catch (LoginException e) {
+            logger.error("SleepIQ cloud login failed", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "SleepIQ cloud login failed");
+        } catch (Exception e) {
+            logger.error("Unexpected error while communicating with SleepIQ cloud", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Unable to connect to SleepIQ cloud");
+        }
+    }
+
+    /**
+     * Create a new SleepIQ cloud service connection. If a connection already exists, it will be lost.
+     *
+     * @throws LoginException if there is an error while authenticating to the service
+     */
+    private void createCloudConnection() throws LoginException {
+
+        logger.debug("Reading SleepIQ cloud binding configuration");
+        SleepIQCloudConfiguration bindingConfig = getConfigAs(SleepIQCloudConfiguration.class);
+
+        logger.debug("Creating SleepIQ client");
+        Configuration cloudConfig = new Configuration().withUsername(bindingConfig.username)
+                .withPassword(bindingConfig.password).withLogging(bindingConfig.logging);
+        cloud = SleepIQ.create(cloudConfig);
+
+        logger.info("Authenticating to SleepIQ cloud service");
+        cloud.login();
+
+        logger.info("Authentication successful");
```

Comment:
If you want to keep that as info level maybe reformulate it to `Successfully authenticated at the SleepIQ cloud service.`